flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -121,18 +115,46 @@ public StreamTwoInputProcessor(
 			taskManagerConfig,
 			taskName);
 		checkState(checkpointedInputGates.length == 2);
-		this.input1 = new StreamTaskNetworkInput(checkpointedInputGates[0], inputSerializer1, ioManager, 0);
-		this.input2 = new StreamTaskNetworkInput(checkpointedInputGates[1], inputSerializer2, ioManager, 1);
 
-		this.statusWatermarkValve1 = new StatusWatermarkValve(
-			unionedInputGate1.getNumberOfInputChannels(),
-			new ForwardingValveOutputHandler(streamOperator, lock, streamStatusMaintainer, input1WatermarkGauge, 0));
-		this.statusWatermarkValve2 = new StatusWatermarkValve(
-			unionedInputGate2.getNumberOfInputChannels(),
-			new ForwardingValveOutputHandler(streamOperator, lock, streamStatusMaintainer, input2WatermarkGauge, 1));
+		this.output1 = new StreamTaskNetworkOutput<>(
+			streamOperator,
+			(StreamRecord<IN1> record) -> {
```

Comment:
Agree, the constructor seems too long now. In order to avoid this, I ever constructed the relevant arguments including `Output` via separate method in `StreamTwoInputTask`, and then pass these final arguments to the constructor of `StreamTwoInputProcessor` directly in my previous version. I would consider refactor this part later.